packages/core/primitives/event-dispatch/src/eventcontract.ts

```diff
@@ -282,9 +284,14 @@ export class EventContract implements UnrenamedEventContract {
     const earlyEventHandler: (event: Event) => void = earlyJsactionData.h;
     for (let idx = 0; idx < earlyEventTypes.length; idx++) {
       const eventType: string = earlyEventTypes[idx];
-      window.document.documentElement.removeEventListener(eventType, earlyEventHandler);
+      earlyJsactionData.c.removeEventListener(eventType, earlyEventHandler);
+    }
+    const earlyEventTypesCapture: string[] = earlyJsactionData.etc;
+    for (let idx = 0; idx < earlyEventTypesCapture.length; idx++) {
+      const eventType: string = earlyEventTypes[idx];
+      earlyJsactionData.c.removeEventListener(eventType, earlyEventHandler);
```

Comment:
Should we add a "capture" argument while removing a listener?
```suggestion
      earlyJsactionData.c.removeEventListener(eventType, earlyEventHandler, true /* useCapture */);
```

Reply:
yes